### PR TITLE
github-management: document team structure and process

### PR DESCRIPTION
**Pull request by nikhita**
Fixes https://github.com/kubernetes/community/issues/3102
and address the docs part of https://github.com/kubernetes/org/issues/563

/kind documentation
/cc @spiffxp @justaugustus @mrbobbytables @idealhack @rlenferink 
/assign @spiffxp

### CODE REVIEW

File: github-management/org-owners-guide.md [hunk 1]
```diff
@@ -76,6 +78,24 @@ for all orgs going forward.  Notable discrepancies at the moment:
   org. This was a mistake that swapped the usual convention, and we would like
   to rename the team
 
+### Structure and process
```
Comment (mrbobbytables) [inline github-management/org-owners-guide.md]:
```suggestion
### Structure and Process
```

Super tiny nit :x Capitalize the P to match other headings.

File: github-management/org-owners-guide.md [hunk 2]
```diff
@@ -76,6 +78,24 @@ for all orgs going forward.  Notable discrepancies at the moment:
...
+  - Note that because of how the GiHub API works, org admins still need to be
+    in the `maintainers` list for the teams they are a part of.
+- The `privacy` of a team *must* be `closed`.
+- A new team can be created or a member can be added to a team by
+creating a PR against the [kubernetes/org] repo. The PR must be
+approved by the relevant `OWNERS` or the SIG leads.
+  - For example, addition of a member to `foo-maintainers` must be approved
+    by the `OWNERS` of the repo `foo` or the leads of the SIG associated
+    with the repo.
+- To rename a team, add the `previously: <old-team-name>` field to the team
```
Comment (justaugustus) [inline github-management/org-owners-guide.md]: I would put this as the first bullet. If someone is renaming a team, it's quicker for them to know this ahead of time before walking through the rest of the instructions.

File: github-management/org-owners-guide.md [hunk 3]
```diff
@@ -76,6 +78,24 @@ for all orgs going forward.  Notable discrepancies at the moment:
   org. This was a mistake that swapped the usual convention, and we would like
   to rename the team
 
+### Structure and Process
+
+Guidelines on how to create and structure teams are described below:
+
+- Teams should not have any `maintainers` and everyone who should be part of
```
Comment (justaugustus) [inline github-management/org-owners-guide.md]: Could we tweak the language here? Teams will still need to have maintainers if the org owners are part of the team, so this is not strictly accurate. Also, we should make it easy for people to discover an org owner who is part of that team, so maybe a link to the @kubernetes/owners team would be helpful.

Comment (rlenferink) [reply github-management/org-owners-guide.md]: We've got the note mentioning that org admins still need to be in the `maintainers` list, I think that's clear enough. But it indeed might be good to link to `@kubernetes/owners` as well

**Comment by rlenferink** [comment]
Just a kind reminder that this PR is still open and we already merged the `k8s/org` changes ;)

**Comment by spiffxp** [comment]
/approve
/lgtm
Let's merge and iterate, if the concerns raised here are still pressing, I will happily review followup PR's

**Pull request merged**